### PR TITLE
🌱Add image validation fuzzing for CAPM3 

### PR DESCRIPTION
**Pull request by smoshiur1237**
The image validation fuzz test functions works in the following way:

```
FuzzImageValidate
├─ Creates test inputs (17 seeds + millions of mutations)
├─ For each input:
     ├─ Build Image struct
     ├─ [Panic Recovery] Call Validate()
     ├─ Check result is well-formed
     └─ Call checkInvariants()
          └─ Verify 4 rules hold true
```

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
/cc @tuminoid @Rozzii

**Comment by tuminoid** [comment]
Taking a look next week.

**Comment by smoshiur1237** [comment]
Addressed all the reviews. @tuminoid @lentzi90 @Rozzii

**Comment by terror96** [comment]
lgtm

File: test/fuzz/image_validate_fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+	infrav1 "github.com/metal3-io/cluster-api-provider-metal3/api/v1beta2"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// FuzzImageValidate tests the Image.Validate function with random inputs
+// to discover edge cases, panics, or unexpected behaviors.
+// This fuzz test aims to ensure that Image.Validate handles all possible
+// string inputs gracefully without crashing, even with malformed URLs,
+// special characters, or unusual input combinations.
```
Comment (tuminoid) [inline test/fuzz/image_validate_fuzz_test.go]: Overcommenting

File: test/fuzz/image_validate_fuzz_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+func FuzzImageValidate(f *testing.F) {
+	// Seed corpus with known valid and invalid cases
+	// These provide starting points for the fuzzer to mutate
```
Comment (tuminoid) [inline test/fuzz/image_validate_fuzz_test.go]: Overcomment

File: test/fuzz/image_validate_fuzz_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	f.Add("http://example.com/image", "", "", false)                   // Missing checksum
+	f.Add("http://example.com", "http://:invalid", "", false)          // Malformed checksum URL
+	f.Add("ftp://example.com/image", "checksum", "", false)            // Non-HTTP protocol
+	f.Add("http://example.com/image", "https://", "", false)           // Incomplete checksum URL
+	f.Add("http://", "checksum", "", false)                            // Incomplete URL
+	f.Add("://example.com", "checksum", "", false)                     // Missing protocol
+	f.Add("http://example.com/\x00", "checksum", "", false)            // Null byte
+	f.Add("http://example.com/image", "http://\x00", "", false)        // Null byte in checksum
+	f.Add(strings.Repeat("a", 10000), "checksum", "", false)           // Very long URL
+	f.Add("http://example.com", strings.Repeat("a", 10000), "", false) // Very long checksum
```
Comment (tuminoid) [inline test/fuzz/image_validate_fuzz_test.go]: I don't think these comments per line are useful either.

**Review by tuminoid — COMMENTED** [review]
Just some nits. Drop the extra commentary and we'll merge.

/approve
/hold

**Review by Rozzii — COMMENTED** [review]
/lgtm

File: test/fuzz/image_validate_fuzz_test.go (hunk 1 above)
Comment (smoshiur1237) [reply test/fuzz/image_validate_fuzz_test.go]: Removed

File: test/fuzz/image_validate_fuzz_test.go (hunk 2 above)
Comment (smoshiur1237) [reply test/fuzz/image_validate_fuzz_test.go]: Removed

File: test/fuzz/image_validate_fuzz_test.go (hunk 3 above)
Comment (smoshiur1237) [reply test/fuzz/image_validate_fuzz_test.go]: Removed

**Comment by smoshiur1237** [comment]
/cc @Rozzii @adilGhaffarDev 
Addressed all the comments. please reivew

**Review by Rozzii — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
/override metal3-centos-e2e-integration-test-main
/override metal3-ubuntu-e2e-integration-test-main

Thanks Moshiur, this is good to go.

/unhold
